python/bind_slice_methods.h

```diff
@@ -60,26 +60,66 @@ template <class View> struct SetData {
   };
 };
 
-// Helpers wrapped in struct to avoid unresolvable overloads.
-template <class T> struct slicer {
-  static auto get_slice(T &self, const std::tuple<Dim, scipp::index> &index) {
-    auto [dim, i] = index;
-    auto sz = dim_extent(self, dim);
-    if (i < -sz || i >= sz) // index is out of range
-      throw std::runtime_error(
-          "The requested index " + std::to_string(i) +
-          " is out of range. Dimension size is " + std::to_string(sz) +
-          " and the allowed range is [" + std::to_string(-sz) + ":" +
-          std::to_string(sz - 1) + "].");
-    if (i < 0)
-      i = sz + i;
-    return Slice(dim, i);
-  }
+template <class T>
+auto get_slice(T &self, const std::tuple<Dim, scipp::index> &index) {
+  auto [dim, i] = index;
+  auto sz = dim_extent(self, dim);
+  if (i < -sz || i >= sz) // index is out of range
+    throw std::runtime_error(
+        "The requested index " + std::to_string(i) +
+        " is out of range. Dimension size is " + std::to_string(sz) +
+        " and the allowed range is [" + std::to_string(-sz) + ":" +
+        std::to_string(sz - 1) + "].");
+  if (i < 0)
+    i = sz + i;
+  return Slice(dim, i);
+}
 
-  static auto get(T &self, const std::tuple<Dim, scipp::index> &index) {
-    return self.slice(get_slice(self, index));
+template <class T>
+auto get_slice_range(T &self, const std::tuple<Dim, const py::slice> &index) {
+  auto [dim, py_slice] = index;
+  if constexpr (std::is_same_v<T, DataArray> || std::is_same_v<T, Dataset>) {
+    try {
+      auto start = py::getattr(py_slice, "start");
+      auto stop = py::getattr(py_slice, "stop");
+      if (!start.is_none() || !stop.is_none()) { // Means default slice : is
+                                                 // treated as index slice
+        auto start_var = start.is_none()
+                             ? Variable{}
+                             : py::getattr(py_slice, "start").cast<Variable>();
+        auto stop_var = stop.is_none()
+                            ? Variable{}
+                            : py::getattr(py_slice, "stop").cast<Variable>();
+        auto step = py::getattr(py_slice, "step");
+        if (!step.is_none()) {
+          throw std::runtime_error(
+              "Step cannot be specified for value based slicing.");
+        }
+        return std::make_from_tuple<Slice>(get_slice_params(
+            self.dims(), self.coords()[dim], start_var, stop_var));
+      }
+    } catch (const py::cast_error &) {
```

Comment:
What can cause this and why do we ignore it?

Reply:
Positional slicing (with integers) rather than variables (for label-based slicing). This is not new, just moved